File: atcoder/go/abc149/d.go
package main

import (
	"fmt"
)

func main() {
	var n, k, r, s, p int
	var st string
	fmt.Scan(&n, &k, &r, &s, &p, &st)
	l := len(st)
	ans := 0
	for i := 0; i < k; i++ {
		idx := i
		cnt := 0
		for idx+k < l {
			if st[idx] == st[idx+k] {
				cnt++
			} else {
				if st[idx] == 'r' {
					ans += p * (cnt/2 + 1)
				} else if st[idx] == 'p' {
					ans += s * (cnt/2 + 1)
				} else {
					ans += r * (cnt/2 + 1)
				}
				cnt = 0
			}
			idx += k
		}
		if st[idx] == 'r' {
			ans += p * (cnt/2 + 1)
		} else if st[idx] == 'p' {
			ans += s * (cnt/2 + 1)
		} else {
			ans += r * (cnt/2 + 1)
		}
	}
	fmt.Println(ans)
}

File: atcoder/python/abc149/d.py
n,k=map(int,input().split())
r,s,p=map(int,input().split())
t=input()
ans=0
for i in range(k):
    x=i
    cnt=0
    while x+k<n:
        if t[x]==t[x+k]:
            cnt+=1
        else:
            if t[x]=="r":
                ans+=p*(cnt//2+1)
            elif t[x]=="s":
                ans+=r*(cnt//2+1)
            else:
                ans+=s*(cnt//2+1)
            cnt=0
        x+=k
    if t[x]=="r":
        ans+=p*(cnt//2+1)
    elif t[x]=="s":
        ans+=r*(cnt//2+1)
    else:
        ans+=s*(cnt//2+1)
print(ans)